src/func.cc

```diff
@@ -946,6 +946,29 @@ void ProfileFunc(const vector<Value*>& args, Evaluator* ev, string*) {
   }
 }
 
+void VariableLocationFunc(const vector<Value*>& args, Evaluator* ev, string* s) {
+  string arg = args[0]->Eval(ev);
+  WordWriter ww(s);
+  for (StringPiece var : WordScanner(arg)) {
+    Symbol sym = Intern(var);
+    Var* v = ev->PeekVar(sym);
+    const Loc& loc = v->Location();
+    if (loc.filename) {
+      ww.Write(loc.filename);
+    } else {
+      ww.Write("<unknown>");
+    }
+    if (loc.lineno > 0) {
+      char buf[20];
+      sprintf(buf, ":%d", loc.lineno);
```

Comment:
Use snprintf here. 19 bytes should be enough for everyone, but better safe than sorry. To be memory-safe why not something like

```
AppendString(buf, ":");
AppendString(bug, std::to_string(loc.lineno > 0 ? loc.lineno : 0)
```

?

Reply:
That works too.